services/agent-router/internal/httpapi/place_test.go

```
@@ -0,0 +1,580 @@
+		}
+		alt := findAlt(t, p.Alternatives, "kserve-a5000")
+		if !alt.Eligible || alt.Reason.Code != "not_selected_colder" {
+			t.Errorf("kserve-a5000 alternative = %+v, want eligible/not_selected_colder", alt)
+		}
+	})
+
+	// cached_vs_absent: cached and absent are distinct costs and distinct
+	// readiness values, never collapsed (owner ruling R11).
+	t.Run("cached_vs_absent", func(t *testing.T) {
+		e := newEnv(t, realCatalogState(t))
+		e.do(t, http.MethodPost, "/v1/capacity/heartbeat", nodeToken,
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)